### PR TITLE
Boyer Moore Algorithm in C#

### LINKED ISSUE #2480: Boyer Moore Algorithm
- [x] C  - @ravireddy0
- [x] C++
- [x] C# - @HardevKhandhar
- [ ] CoffeeScript
- [ ] Dart
- [x] Go - @jedicakelord
- [x] Java
- [ ] JS - @jaassmine
- [x] Kotlin - @darkmatter18
- [x] Php - @sukritishah15
- [x] Python
- [x] Readme - @simrann20
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by HardevKhandhar**
Fixes #2480 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Boyer Moore Algorithm in C#
- 

#### Languages Used:
- C#
- 

#### Files Added:
- Boyer_Moore.cs
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Boyer_Moore_Algorithm/Boyer_Moore.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        int m = pat.Length; 
+        int n = txt.Length;
+
+        int []Character = new int[CHARACTERS];
+        badChar(pat, m, Character); 
+
+        /* 
+            s is used to keep track of 
+            pattern shifting with respect to text
+	*/
```
Comment (asha15) [inline Boyer_Moore_Algorithm/Boyer_Moore.cs:43]: correct the indentation

File: Boyer_Moore_Algorithm/Boyer_Moore.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
+/**
+    Code contributed by Hardev Khandhar for GirlScript Summer of Code 2020
+    link: https://github.com/HardevKhandhar
+*/
+
```
Comment (asha15) [inline Boyer_Moore_Algorithm/Boyer_Moore.cs:5]: No need this line @HardevKhandhar . it can track through git log. shall we remove this?

**Comment by HardevKhandhar** [comment]
Dear Ma'am @asha15 
The suggested changes are made & the file is updated.
The commits are squashed.
Thank you.

**Comment by asha15** [comment]
Keep going Hardev!

**Pull request merged**